### DISCUSSION TITLE
Welcome to MCPIL Discussions!

### DISCUSSION THREAD

**Discussion by torralbaa** [MCPI-Devs/MCPIL#13] [General]
## :wave: Welcome!
We will be using Discussions as a "forum"/Q&A system. So this is aimed to be a more free and less formal place than issues. Feel free to:
 + Ask questions.
 + Share ideas.
 + Ask for feedback.
 + Publish your project.

However, please keep the Discussions on-topic (MCPIL-related) and follow the rules:
 + Do not insult, threaten, harass, or flame other users. This includes racism, sexism and homophobic behaviour.
 + Be respectful to all the other users.
 + Do not post links to illegal or dangerous websites.
 + Do not share or ask for personal or sensitive data. This includes phone number, address, etc.
 + You must be over the age of 13.
 + No NSFW content.
 + Don't ping/mention roles/everyone/here.
 + Do not swear/say "bad words".

_Rules are subject to common sense._

**Comment by StealthHydra179** [comment]
when did this come out?

**Reply by torralbaa** [reply]
I don't know, it's a Beta feature, but it is awesome!

**Reply by MMK21Hub** [reply]
It launched to all repositories yesterday (along with Dark Theme). I know that https://github.com/eslint/eslint had it for a while before that. You'll probably be able to find some blog post on it.

**Comment by ForsythsGambit** [comment]
This is looking really cool!

**Reply by torralbaa** [reply]
Thank you for your feedback! :D

**Comment by AprentiUnity** [comment]
Hello,
I wanted to know how do you modify Minecraft pi edition with mods in .mcpi extension? Because, for now, all I've seen for modifying Minecraft pi editon are .bsdiff's designed to modify the original file.

**Reply by torralbaa** [reply]
Hello @AprentiUnity!

`.mcpi` mods are just Python mods compressed with a common algorithm (GZip, DEFLATE). So you could write any Python mod using the API, ModPi, and the standard library, and then "compile" (compress) it to make it load-able by MCPIL.

**:warning: However**, be warned: MCPIL is currently dead, please use [MCPIL-R](https://github.com/MCPI-Devs/MCPIL-R), it does not support Python mod loading yet, but has some cool MCPI-Docker mods.

There's more info in the [README](https://github.com/MCPI-Devs/MCPIL#README), but feel free to ask any questions!
